### PR TITLE
Allow shipit + automerge on deletions from ignore.txt files

### PR DESCRIPTION
**Pull request by jctanner**
https://github.com/ansible/ansibullbot/issues/1239
https://github.com/ansible/ansibullbot/issues/1238

### CODE REVIEW

File: ansibullbot/triagers/plugins/shipit.py [hunk 1]
```diff
@@ -261,6 +261,14 @@ def get_shipit_facts(issuewrapper, inmeta, module_indexer, core_team=[], botname
     ]
 
     files = [f for f in iw.files if not f.startswith(u'changelogs/fragments/')]
+
+    # https://github.com/ansible/ansibullbot/issues/1238
+    meta[u'component_matches'] = [
+        x for x in meta.get(u'component_matches', [])
+        if not x[u'repo_filename'].startswith(u'test/sanity')
+    ]
+    files = [f for f in files if not f.startswith(u'test/sanity') and f.endswith(u'ignore.txt')]
```
Comment (mkrizek) [inline ansibullbot/triagers/plugins/shipit.py]:
```suggestion
    files = [f for f in files if not (f.startswith(u'test/sanity') and f.endswith(u'ignore.txt'))]
```

File: ansibullbot/triagers/plugins/shipit.py [hunk 2]
```diff
@@ -261,6 +261,14 @@ def get_shipit_facts(issuewrapper, inmeta, module_indexer, core_team=[], botname
     ]
 
     files = [f for f in iw.files if not f.startswith(u'changelogs/fragments/')]
+
```
Comment (mkrizek) [inline ansibullbot/triagers/plugins/shipit.py]: https://github.com/ansible/ansibullbot/blob/a06dab3fb4b710fb38cc4b09894900a8f6bb8467/ansibullbot/triagers/plugins/shipit.py#L113 Shouldn't the above be also changed to allow for `test/sanity/ignore.txt`? Won't it fail with `automerge !module file(s) test failed` currently?

File: ansibullbot/triagers/plugins/shipit.py [hunk 3]
```diff
@@ -261,6 +261,14 @@ def get_shipit_facts(issuewrapper, inmeta, module_indexer, core_team=[], botname
...
+    # https://github.com/ansible/ansibullbot/issues/1238
+    meta[u'component_matches'] = [
+        x for x in meta.get(u'component_matches', [])
+        if not x[u'repo_filename'].startswith(u'test/sanity')
+    ]
+    files = [f for f in files if not (f.startswith(u'test/sanity') and f.endswith(u'ignore.txt'))]
```
Comment (mattclay) [inline ansibullbot/triagers/plugins/shipit.py]: What makes this only work for deletions to these files and not additions/changes?

File: tests/unit/triagers/plugins/test_shipit.py [hunk 4]
```diff
@@ -297,6 +300,33 @@ def test_supershipit_changelogs(self):
         assert sfacts[u'shipit_actors'] == []
         assert sfacts[u'shipit_actors_other'] == [u'jane']
 
+    def test_supershipit_deletion_from_sanity_ignore(self):
```
Comment (mattclay) [inline tests/unit/triagers/plugins/test_shipit.py]: Do we have tests to make sure that supershipit won't work when ignore files are added or have lines added/changed?

File: ansibullbot/triagers/plugins/shipit.py (hunk 3 above)
Comment (jctanner) [reply ansibullbot/triagers/plugins/shipit.py]: Evaluating additions vs deletions is done in the automerge inspection. This function wasn't well designed for the increasing scope so i'm thinking about a better way to refactor it all.

File: tests/unit/triagers/plugins/test_shipit.py (hunk 4 above)
Comment (jctanner) [reply tests/unit/triagers/plugins/test_shipit.py]: TBD

**Comment by xuxiaowei0512** [comment]
This will be a good thing. 
What about 'test/sanity/pylint/ignore.txt'?
Is this included?

File: ansibullbot/triagers/plugins/shipit.py [hunk 5]
```diff
@@ -261,6 +261,21 @@ def get_shipit_facts(issuewrapper, inmeta, module_indexer, core_team=[], botname
...
+
+    # make sure only deletions from ignore.txt are allowed
+    for pr_file in iw.pr_files:
+        if (pr_file.filename.startswith(u'test/sanity') and pr_file.filename.endswith(u'ignore.txt')):
+            if pr_file.additions > 0:
```
Comment (mattclay) [inline ansibullbot/triagers/plugins/shipit.py]: Do we need a check for added files? The existing code checking for deletions has this:

https://github.com/ansible/ansibullbot/blob/a6e389c92b786e803df7c4bd88c6715257c065eb/ansibullbot/triagers/plugins/shipit.py#L114

File: tests/unit/triagers/plugins/test_shipit.py [hunk 6]
```diff
@@ -975,3 +1069,74 @@ def test_automerge_changelog_fragment(self):
...
+
+        afacts = get_automerge_facts(iw, meta)
+
+        self.assertTrue(afacts[u'automerge'])
+
+    def test_automerge_addition_to_ignore(self):
+        iw = IssueWrapperMock('ansible', 'ansible', 1)
+        iw._is_pullrequest = True
+        mfile = MockFile(u'test/sanity/validate-modules/ignore.txt')
+        mfile.addtions = 1
```
Comment (mattclay) [inline tests/unit/triagers/plugins/test_shipit.py]: Typo?

```suggestion
        mfile.additions = 1
```

Also, if this test is passing now, that would seem to indicate we're not testing what we think we are.

**Comment by jctanner** [comment]
superceded by https://github.com/ansible/ansibullbot/pull/1275

**Pull request closed without merging**